fud/fud/json_to_dat.py

```diff
@@ -1,21 +1,197 @@
 import json
 import numpy as np
 from pathlib import Path
-
+from .errors import ConversionError
 
 # Converts `val` into a bitstring that is `bw` characters wide
 def bitstring(val, bw):
     # first truncate val by `bw` bits
     val &= (2**bw - 1)
     return '{:x}'.format(val)
 
+# Converts `val` in binary representation to a bitstring that is `bw` characters wide
+def bitstring2(val, bw):
+    #first truncate val by `bw` bits
+    val = val[len(val)-bw:len(val)]
+    numval = int(val,base=2)
+    return '{:x}'.format(numval)
 
 def parse_dat(path):
     with path.open('r') as f:
         lines = f.readlines()
         arr = np.array(list(map(lambda v: int(v.strip(), 16), lines)))
         return arr
 
+def signed_parse_helper(v,bw):
```

Comment:
can you document this function?

Reply:
# since the number should be signed, if the value v is v > 2**(bw-1), the number should be translated as a negative number